nav2_bt_navigator/CMakeLists.txt

```diff
@@ -17,6 +17,8 @@ find_package(nav2_tasks REQUIRED)
 find_package(nav2_msgs REQUIRED)
 find_package(nav2_robot REQUIRED)
 find_package(behavior_tree_core REQUIRED)
+find_package(tf2 REQUIRED)
```

Comment:
Why does the BT Navigator required additional packages? Should just be updating the XML and registering new types.

Reply:
It should, the tf2 is used by `nav2_tasks::spin_action` I tried adding it to the CMake of `nav2_tasks` however I get some linking errors. Can you help me take a look?